torch/fx/experimental/symbolic_shapes.py

```diff
@@ -348,6 +336,24 @@ def constrain_range(a, *, min: Optional[int], max: Optional[int] = None):
         builtins.max(r.lower, min), builtins.min(r.upper, max)
     )
 
+def constrain_range_non_symint(a, *, min, max):
+    assert not isinstance(a, SymInt)
+    if not (min <= a <= max):
+        raise ValueRangeError(f"Invalid value {a} for range [{min}:{max}]")
+
+    if (
+        (fake_mode := detect_fake_mode()) is not None and
+        getattr(fake_mode, "shape_env", None) is not None
+    ):
+        # If we are tracing with a fake mode then add this integer to the
+        # shape_env's var_to_range
+        sym_integer = sympy.Integer(a)
+        shape_env = fake_mode.shape_env
+        shape_env.var_to_range[sym_integer] = ValueRanges(min, max)
+        shape_env.var_to_stack[sym_integer] = TracingContext(fake_mode).extract_stack()
+
+    return
```

Comment:
delete

Reply:
Let me follow up with @tugsbayasgalan and @angelayi for some context of this logic.

Reply:
OK, thanks. Just to be clear, here where I said delete I just mean the `return`, because it's superfluous. But yeah, good idea to get some context...I'm not sure myself when this or the other code path is taken (something is a symint vs. not).